pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -159,6 +164,33 @@ func completeProducerJob(
 	}
 }
 
+// startPostCutoverRetentionJob begins a dummy producer job on the newly cutover
+// to tenant. This producer job will lay PTS over the whole tenant, enabling a
+// fast failback to the original source cluster.
+//
+// TODO (msbutler): currently the PTS expires after
+// stream_replication.job_liveness_timeout time elapses. This should be user
+// configurable via the cutover command.
+func startPostCutoverRetentionJob(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	details jobspb.StreamIngestionDetails,
+	cutoverTime hlc.Timestamp,
+) error {
+
+	req := streampb.ReplicationProducerRequest{
+		ReplicationStartTime: cutoverTime,
+		CutoverRetention:     true,
+	}
+	reqBytes, err := protoutil.Marshal(&req)
+	if err != nil {
+		return err
+	}
+	iExec := execCfg.InternalDB.Executor()
```

Comment:
I think we could just call the go function instead detouring though sql builtin?

Reply:
So, i tried doing this with this [additional commit](https://github.com/cockroachdb/cockroach/commit/f43d82f5a356deb710a10fdbda7dd6243259df0e), but `streamFactory.GetReplicationStreamManager(ctx)` returns the error `cannot use HasAdminRole without a txn`, I think because the `evalCtx` hanging off `execCtx.ExtendedEvalContext()` is not associated with a txn. Do you have any suggestions for how to create an ExentedEvalCtx() with a txn here? I can also just keep the workaround by using the builtin.

Reply:
Don’t use the manager? I’d pull the logic into a function here in `streamingest` that is shared by the manager and this code

Reply:
oh duh. Done.